File: selection-sort.js
let unsorted = [5, 2, 9, 1, 11, 7, 8, -1, 4, 3, 10, 14, 0, 6];

const selection_sort = (unsorted) => {

    for (let i = 0; i < unsorted.length; i++) {

        let smallest_index = i;

        for (let j = i + 1; j < unsorted.length; j++) {

            if (unsorted[j] < unsorted[smallest_index]) {
                smallest_index = j;
            }
        }

        let temp = unsorted[smallest_index];
        unsorted[smallest_index] = unsorted[i];
        unsorted[i] = temp;

    }

    return unsorted;

}

console.log(selection_sort(unsorted));

File: sort/sort/SelectionSort.java
package sort;

public class SelectionSort {

    public static void main(String[] args) {

        int[] sorted = selectionSort(new int[] { 3, 5, 1, 6, 2, 7, 4, 8, 9 });

        for (int i : sorted) {
            System.out.println(i);
        }
    }
    
    public static int[] selectionSort(int[] unsorted) {

        for (int i = 0; i < unsorted.length - 1; i++) {

            int min = i;

            for (int j = i + 1; j < unsorted.length; j++) {

                if (unsorted[j] < unsorted[min]) {
                    min = j;
                }
            }

            int temp = unsorted[min];
            unsorted[min] = unsorted[i];
            unsorted[i] = temp;
        }
        
        return unsorted;
    }
}